src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/Extensions/IVerifierExtensions.cs

```diff
@@ -56,5 +68,80 @@ public static void EqualOrDiff(this IVerifier verifier, string expected, string
+        {
+            private const string CarriageReturnText = "<CR>";
+            private const string LineFeedText = "<LF>";
+
+            private static readonly char[] s_endOfLineCharacters = { '\r', '\n' };
+            private static readonly Differ s_differ = new Differ();
+
+            public DiffResult CreateCharacterDiffs(string oldText, string newText, bool ignoreWhitespace)
+                => s_differ.CreateCharacterDiffs(oldText, newText, ignoreWhitespace);
+
+            public DiffResult CreateCharacterDiffs(string oldText, string newText, bool ignoreWhitespace, bool ignoreCase)
+                => s_differ.CreateCharacterDiffs(oldText, newText, ignoreWhitespace, ignoreCase);
+
+            public DiffResult CreateCustomDiffs(string oldText, string newText, bool ignoreWhiteSpace, Func<string, string[]> chunker)
+                => s_differ.CreateCustomDiffs(oldText, newText, ignoreWhiteSpace, chunker);
+
+            public DiffResult CreateCustomDiffs(string oldText, string newText, bool ignoreWhiteSpace, bool ignoreCase, Func<string, string[]> chunker)
+                => s_differ.CreateCustomDiffs(oldText, newText, ignoreWhiteSpace, ignoreCase, chunker);
+
+            public DiffResult CreateLineDiffs(string oldText, string newText, bool ignoreWhitespace)
+                => CreateLineDiffs(oldText, newText, ignoreWhitespace, ignoreCase: false);
+
+            public DiffResult CreateLineDiffs(string oldText, string newText, bool ignoreWhitespace, bool ignoreCase)
+            {
+                Func<string, string[]> chunker = s =>
+                {
+                    var lines = new List<string>();
+
+                    var nextChar = 0;
+                    while (nextChar < s.Length)
+                    {
+                        var nextEol = s.IndexOfAny(s_endOfLineCharacters, nextChar);
+                        if (nextEol == -1)
+                        {
+                            lines.Add(s.Substring(nextChar));
+                            break;
+                        }
+
+                        var currentLine = s.Substring(nextChar, nextEol - nextChar);
+
+                        switch (s[nextEol])
+                        {
+                            case '\r':
+                                currentLine += CarriageReturnText;
+                                if (nextEol < s.Length - 1 && s[nextEol + 1] == '\n')
+                                {
+                                    currentLine += LineFeedText;
+                                    nextEol++;
+                                }
+
+                                break;
+
+                            case '\n':
+                                currentLine += LineFeedText;
+                                break;
+                        }
+
+                        lines.Add(currentLine);
+                        nextChar = nextEol + 1;
+                    }
+
+                    return lines.ToArray();
```

Comment:
didn't review.  i trust you :)

Reply:
I made a mistake originally but the tests caught it 👍